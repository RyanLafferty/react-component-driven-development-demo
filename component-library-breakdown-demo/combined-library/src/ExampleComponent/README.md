### Usage
```js
import { ExampleComponent } from 'combined-library';

...

<ExampleComponent />
```
