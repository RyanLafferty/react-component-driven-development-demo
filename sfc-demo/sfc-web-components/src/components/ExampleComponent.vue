<template>
  <div class="container" />
</template>

<script>
export default {
  name: 'ExampleComponent',
};
</script>

<style scoped lang="scss">
  .container {
    background-color: #f036f0;
    border-radius: 50%;
    height: 200px;
    width: 200px;
  };
</style>
