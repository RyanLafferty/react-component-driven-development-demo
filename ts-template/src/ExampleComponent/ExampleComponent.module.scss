.container {
  color: Green;
  font-size: 16px;
};
