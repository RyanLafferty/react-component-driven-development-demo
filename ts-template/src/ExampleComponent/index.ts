export { default } from './ExampleComponent';
export { ExampleComponentProps } from './ExampleComponent.types';
