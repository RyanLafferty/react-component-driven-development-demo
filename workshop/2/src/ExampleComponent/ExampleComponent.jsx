import React from 'react';
import PropTypes from 'prop-types';

const ExampleComponent = ({ name }) => (
  <div>
    {`Hello ${name}!`}
  </div>
);

ExampleComponent.propTypes = {
  name: PropTypes.string,
};

ExampleComponent.defaultProps = {
  name: 'World',
};

export default ExampleComponent;
