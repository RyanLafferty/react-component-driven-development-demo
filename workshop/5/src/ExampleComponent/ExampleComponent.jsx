import React from 'react';
import PropTypes from 'prop-types';
import styles from './ExampleComponent.module.scss';

const ExampleComponent = ({ name }) => (
  <div className={styles.message}>
    {`Hello ${name}!`}
  </div>
);

ExampleComponent.propTypes = {
  name: PropTypes.string,
};

ExampleComponent.defaultProps = {
  name: 'World',
};

export default ExampleComponent;
