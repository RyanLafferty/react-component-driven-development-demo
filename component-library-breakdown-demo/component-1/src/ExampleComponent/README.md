### Usage
```js
import ExampleComponent from 'component-1';

...

<ExampleComponent />
```
