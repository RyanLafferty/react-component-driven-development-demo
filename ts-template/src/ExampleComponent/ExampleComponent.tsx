import React from 'react';
import styles from './ExampleComponent.module.scss';
import { ExampleComponentProps } from './ExampleComponent.types';

const ExampleComponent: React.FC<ExampleComponentProps> = ({ name }) => (
  <div className={styles.container}>
    Hello {name}!
  </div>
);

export default ExampleComponent;
