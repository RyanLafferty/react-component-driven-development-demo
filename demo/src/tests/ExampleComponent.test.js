import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer from 'react-test-renderer';
import ExampleComponent from 'ExampleComponent';

describe('ExampleComponent', () => {
  let props;
  const renderer = new ShallowRenderer();

  beforeEach(() => {
    props = {
      onClick: jest.fn(),
    };
  });

  describe('when no props are provided', () => {
    it('ExampleComponent renders', () => {
      expect(
        renderer.render(<ExampleComponent {...props} />),
      ).toMatchSnapshot();
    });
  });

  describe('when ExampleComponent is clicked', () => {
    it('calls onClick', () => {
      const component = TestRenderer.create(<ExampleComponent {...props} />);
      const button = component.root.findByProps({role: 'button'});

      button.props.onClick();
      expect(props.onClick).toHaveBeenCalled();
    });
  });

  describe('when ExampleComponent is pressed', () => {
    it('calls onClick', () => {
      const component = TestRenderer.create(<ExampleComponent {...props} />);
      const button = component.root.findByProps({role: 'button'});

      button.props.onKeyPress();
      expect(props.onClick).toHaveBeenCalled();
    });
  });
});
