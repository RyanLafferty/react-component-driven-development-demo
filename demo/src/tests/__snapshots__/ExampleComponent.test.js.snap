// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleComponent when no props are provided ExampleComponent renders 1`] = `
<div
  aria-label="ExampleComponent"
  className="container"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="0"
/>
`;
